lib/ecto/type.ex

```diff
@@ -434,71 +461,89 @@ defmodule Ecto.Type do
       iex> load(:integer, "10")
       :error
 
-  A `loader` function may be given for handling recursive types.
+  A `loader` function may be given for handling recursive types, see: `load/3`.
   """
-  @spec load(t, term, (t, term -> {:ok, term} | :error)) :: {:ok, term} | :error
-  def load(type, value, loader \\ &load/2)
-
-  def load({:embed, embed}, value, loader) do
-    load_embed(embed, value, loader)
-  end
-
-  def load(_type, nil, _loader), do: {:ok, nil}
-
-  def load(:binary_id, value, _loader) when is_binary(value) do
-    {:ok, value}
-  end
-
-  def load({:array, type}, value, loader) when is_list(value) do
-    array(value, type, loader, [])
-  end
-
-  def load({:map, type}, value, loader) when is_map(value) do
-    map(Map.to_list(value), type, loader, %{})
-  end
-
-  def load(:time, %Time{} = time, _loader) do
-    {:ok, truncate_usec(time)}
-  end
-
-  def load(:time_usec, %Time{} = time, _loader) do
-    {:ok, pad_usec(time)}
+  @spec load(t, term) :: {:ok, term} | :error
+  def load({:embed, embed}, value) do
+    load_embed(embed, value, &load/2)
   end
 
-  def load(:naive_datetime, %NaiveDateTime{} = naive_datetime, _loader) do
-    {:ok, truncate_usec(naive_datetime)}
-  end
-
-  def load(:naive_datetime_usec, %NaiveDateTime{} = naive_datetime, _loader) do
-    {:ok, pad_usec(naive_datetime)}
+  def load(_type, nil) do
+    {:ok, nil}
   end
 
-  def load(:utc_datetime, %NaiveDateTime{} = naive_datetime, _loader) do
-    naive_datetime |> truncate_usec() |> DateTime.from_naive("Etc/UTC")
+  def load(type, value) do
+    load_fun(type).(value)
   end
 
-  def load(:utc_datetime, %DateTime{} = datetime, _loader) do
-    {:ok, truncate_usec(datetime)}
-  end
+  @doc """
+  Dumps a value to the given type using a custom function.
```

Comment:
I think this is a copy & paste error. This should be something like “Loads a value...” and reference `load/2` instead of `dump/2` :)